index.js

```diff
@@ -18,6 +104,17 @@ const oauthPlugin = fp(function (fastify, options, next) {
   next()
 })
 
+function promisify (func) {
+  return function (arg1) {
+    return new Promise(function (resolve, reject) {
+      func(arg1, function (err, token) {
+        if (err) reject(err)
+        else resolve(token)
+      })
+    })
+  }
+}
```

Comment:
You can use https://www.npmjs.com/package/util.promisify instead.

Reply:
done. es6-promisify used